l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -0,0 +1,678 @@
+
+        (address _l1Token, uint256 _depositAmount, address _l2Receiver) = abi.decode(
+            _data,
+            (address, uint256, address)
+        );
+        require(_l1Token != l1WethAddress, "ShB: WETH deposit not supported");
+        require(bridgehub.baseToken(_chainId) != _l1Token, "ShB: baseToken deposit not supported");
+
+        uint256 amount;
+        if (_l1Token == ETH_TOKEN_ADDRESS) {
+            amount = msg.value;
+            require(_depositAmount == 0, "ShB wrong withdraw amount");
+        } else {
+            require(msg.value == 0, "ShB m.v > 0 for BH d.it 2");
+            amount = _depositAmount;
+
+            /// This breaks the _depositeFunds function, it returns 0, as we are withdrawing funds from ourselves, so our balance doesn't increase
+            /// This should not happen, this bridge only calls the Bridgehub if Eth is the baseToken or for wrapped base token deposits
+            require(_prevMsgSender != address(this), "ShB calling itself");
+
+            uint256 withdrawAmount = _depositFunds(_prevMsgSender, _l1Token, _depositAmount);
+            require(withdrawAmount == _depositAmount, "5T"); // The token has non-standard transfer logic
+        }
+        require(amount != 0, "6T"); // empty deposit amount
+
+        bytes32 txDataHash = keccak256(abi.encode(_prevMsgSender, _l1Token, amount));
+        if (!hyperbridgingEnabled[_chainId]) {
+            chainBalance[_chainId][_l1Token] += amount;
+        }
+
+        {
+            // Request the finalization of the deposit on the L2 side
+            bytes memory l2TxCalldata = _getDepositL2Calldata(_chainId, _prevMsgSender, _l2Receiver, _l1Token, amount);
+
+            request = L2TransactionRequestTwoBridgesInner({
+                magicValue: TWO_BRIDGES_MAGIC_VALUE,
+                l2Contract: l2BridgeAddress[_chainId],
+                l2Calldata: l2TxCalldata,
+                factoryDeps: new bytes[](0),
+                txDataHash: txDataHash
+            });
+        }
+        emit BridgehubDepositInitiatedSharedBridge(_chainId, txDataHash, _prevMsgSender, _l2Receiver, _l1Token, amount);
+        // kl todo. We are breaking the previous events here, as we don't have the txHash, so we can't emit a DepositInitiated event
+    }
+
+    /// @notice used by requestL2TransactionTwoBridges in Bridgehub
+    /// used to confirm that the Mailbox has accepted a transaction.
+    /// we can store the fact that the tx has happened using txDataHash and txHash
+    function bridgehubConfirmL2Transaction(
+        uint256 _chainId,
+        bytes32 _txDataHash,
+        bytes32 _txHash
+    ) external override onlyBridgehub {
+        if (_txDataHash != 0x00) {
+            require(depositHappened[_chainId][_txHash] == 0x00, "ShB tx hap");
+            depositHappened[_chainId][_txHash] = _txDataHash;
+            emit BridgehubDepositFinalized(_chainId, _txDataHash, _txHash);
+        }
```

Comment:
Why do you check `_txDataHash != 0x00` but not enforce it? 

In the `bridgehubDeposit` it is always
```
bytes32 txDataHash = keccak256(abi.encode(_prevMsgSender, _l1Token, amount));
```